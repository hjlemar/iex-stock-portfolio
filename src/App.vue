<template>
  <v-app>
    <v-navigation-drawer fixed v-model="drawer" app>
      <app-date-picker></app-date-picker>
      <app-stock-list :stocks='stocks'></app-stock-list>
    </v-navigation-drawer>
    <v-toolbar color="indigo" dark fixed app>
      <v-toolbar-side-icon @click.stop="drawer = !drawer"></v-toolbar-side-icon>
      <v-toolbar-title>Application</v-toolbar-title>
    </v-toolbar>
    <v-content>
      <v-container fluid>
        <v-layout justify-center align-center>
          <app-home></app-home>
        </v-layout>
      </v-container>
    </v-content>
    <v-footer color="indigo" app>
      <span class="white--text">&copy; 2017</span>
    </v-footer>
  </v-app>
</template>

<script>
import { mapGetters } from 'vuex';
import DatePicker from '@/components/DatePicker';
import StockList from '@/components/StockList';
import Home from '@/components/Home';

export default {
  name: 'App',
  components: {
    appDatePicker: DatePicker,
    appStockList: StockList,
    appHome: Home,
  },
  data: () => ({
    drawer: null,
  }),
  computed: {
    ...mapGetters(['stocks']),
  },
};
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  color: #2c3e50;
  margin-top: 20px;
}
</style>
