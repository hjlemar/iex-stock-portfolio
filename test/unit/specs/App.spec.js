import App from '@/App';
import Vuex from 'vuex';
import { shallow, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';


const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('App.vue', () => {
  let store;

  beforeEach(() => {
    const getters = {
      stocks: jest.fn(),
    };
    store = new Vuex.Store({
      state: { },
      getters,
    });
  });

  it('should render component and match snapshot', () => {
    const wrapper = shallow(App, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
