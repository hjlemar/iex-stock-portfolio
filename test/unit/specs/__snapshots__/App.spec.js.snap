// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render component and match snapshot 1`] = `undefined`;
